src/opt/ReplaceMaskedMemOps.cpp

```diff
@@ -0,0 +1,253 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "ReplaceMaskedMemOps.h"
+
+#include <llvm/IR/IRBuilder.h>
+#include <llvm/Transforms/Utils/Local.h>
+
+#include <numeric>
+
+namespace ispc {
+
+bool lIsPowerOf2(unsigned n) { return (n > 0) && !(n & (n - 1)); }
```

Comment:
There are such functions in LLVM, but it would need additional headers and libs to link.